File: frontend/src/model/Assignments.ts
import {UserView} from "./User";
import {Semester} from "./SchoolPeriod";
import {AssignmentValue} from "./AssignmentValue";

export interface Assignments {
    id: number,
    createdBy: UserView,
    createdOn: Date | null
    text: String,
    semester: Semester,
    assignmentType: AssignmentType,
    assignmentValue: AssignmentValue
}


export enum AssignmentType {
    EXAMINATION = "EXAMINATION", HOMEWORK = "HOMEWORK", EVENT = "EVENT"
}

export const assignmentTypePluralTranslation = {
    EXAMINATION: "Контролни работи", HOMEWORK: "Домашни работи", EVENT: "Събития"
}

export const assignmentTypeTranslation = {
    EXAMINATION: "Контролна работа", HOMEWORK: "Домашна работа", EVENT: "Събитие"
}

File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/assignments/Assignments.kt
@file:UseSerializers(
    LocalDateTimeSerializer::class
)

package com.nevexis.backend.schoolManagement.assignments

import com.nevexis.backend.schoolManagement.school_period.Semester
import com.nevexis.backend.schoolManagement.users.UserView
import com.nevexis.backend.serializers.LocalDateTimeSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import java.time.LocalDateTime

@Serializable
data class Assignments(
    val id: Int? = null,
    val createdBy: UserView,
    val createdOn: LocalDateTime? = null,
    val text: String,
    val semester: Semester,
    val assignmentType: AssignmentType,
    val assignmentValue: AssignmentValue
)


enum class AssignmentType {
    EXAMINATION, HOMEWORK, EVENT
}